torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -957,6 +957,31 @@ int64_t normalizeIndex(int64_t idx, int64_t list_size) {
   return idx;
 }
 
+int stringSlice(Stack& stack) {
+  auto step = pop(stack).toInt();
+  AT_CHECK(step == 1, "Slicing a string only supports step=1");
+
+  auto end = pop(stack).toInt();
+  auto start = pop(stack).toInt();
+  auto string = pop(stack).toStringRef();
+  const int64_t size = string.size();
+
+  // Clamp start and end to the bounds of the list
+  start = std::max(int64_t(0), normalizeIndex(start, size));
+  end = std::min(size, normalizeIndex(end, size));
+
+  if (end <= start) {
+    // Slice is empty
+    push(stack, std::string(""));
+    return 0;
+  }
+
+  std::string result;
+  result.insert(result.begin(), string.begin() + start, string.begin() + end);
```

Comment:
you don't need an extra `insert()`, just use the constructor e.g. `std::string result(string, start, end-start)`